### ISSUE TITLE
Propose APK repository to test decompilation and recompilation 

### ISSUE THREAD

**Issue by mingyuan-xia** [skylot/jadx#490]
The current travis config only tests building jadx. The deompilation logic is not tested.
The idea is to set up an APK repo and test if the nightly unstable build can decompile every APK in the repo correctly.

Initially 10-12 APKs, very simple ones (like gallery, sample applications, ones using lambda, kotlin ones, anonymous classes, etc) and slightly bigger one. We can add later on.

For each APK, the test is simple: `jadx-cli` to decompile it, then save the gradle project and recompile the project with java compiler. See if the decompiled code has correct syntax.

**Comment by skylot** [comment]
Hm, so you call 350 tests (240 are integration) in jadx-core - "decompilation logic is not tested"? This is cruel ((

Anyway, recompilation is a tough part, because most apks decompiles with errors now, and most errors occur in common libraries which you will remove and replace with dependency for recompilation.
Also what the purpose of these apks? Catch regression? For that, I will better add one more test, because tests are easier to debug, maintain, etc.

I have tried an approach similar to your suggestion on a simple app I made especially for recompilation test (see [jadx-test-app](https://github.com/skylot/jadx-test-app)) but it is not very useful and need a lot of efforts to keep it up to date to build.

**Comment by mingyuan-xia** [comment]
@_@ Sorry to miss your (quite large body of) unit tests. The recompilation test(s) aim not to reproduce a functionally equivalent recompiled APK, but to let javac check the syntax of jadx-generated Java code. Consider this more as an end-to-end test where all lambda/kotlin/renaming/import/deobfuscation modules work well enough for complicated real-world APKs. We can start easy with APK selection and tolerate decompilation failures(inconsistent code). Actually, the picked APKs in the repo will be some kind of trophy of "the-perfectly-decompiled-samples".
Expected output per APK:
```
Decompiled SLOC: 4,500
Bad decompilation: 15 method body(s)
com.xx.bb.Test.foo()
...
Renamed classes: 230
```
Upon a new release, probably a histogram chart for bad decompilation can show off how the new version reduces errors.

**Comment by skylot** [comment]
As a note: javac checks already executed in integration tests and also execute tests in classes compiled from jadx-generated code for check code equivalence. For example, if you run [this](https://github.com/skylot/jadx/blob/d586c84b5632be0bca22250a7081506136e8a85e/jadx-core/src/test/java/jadx/tests/integration/conditions/TestConditions16.java) test, you will see output:
```java
-----------------------------------------------------------
package jadx.tests.integration.conditions;

import org.junit.Assert;

public class TestConditions16$TestCls {
    private static boolean test(int a, int b) {
        return a < 0 || ((b % 2 != 0 && a > 28) || b < 0);
    }

    public void check() {
        Assert.assertTrue(test(-1, 1));
        Assert.assertTrue(test(1, -1));
        Assert.assertTrue(test(29, 3));
        Assert.assertFalse(test(2, 2));
    }
}
-----------------------------------------------------------
Compilation: PASSED
Auto check: PASSED
```
Here `Compilation: PASSED` means that generated code compiled without errors and `Auto check: PASSED` means that executing `check` method on generated code also succesful.
So, these tests are very strict. Also, you can write tests using smali, for cases then java compiler can't produce desired bytecode.

As I understand, you want to collect statistics to compare jadx releases?
`jadx-cli` already output errors and warnings count, also errors counts returned as execution result. Using this you can write a simple script and collect such stats if you want.
But, let me tell you a secret: you will get more errors with each release for some apks ))
Because I constantly add more checks to mark incorrectly generated code, so if in previous release you see some code but in next release don't, it means previous release produce incorrect code and for me this is unacceptable. This sound strange, but checking code without actual execution is rather hard.
(side note: if someone knows a library which can tell if two java methods are equivalent, please tell me)

In short: I inderstand why you want this "apk testing facility" but I don't think it must be part of a build, because it will be very slow and resource consuming and also will not give any improvement in jadx development. You can run jadx youself on any apks you want and share any interesting stats or issues.
